agent/requestor/src/main.rs

```diff
@@ -195,20 +192,38 @@ async fn main() -> anyhow::Result<()> {
         payment_api.clone(),
         started_at,
         agreement_allocation.clone(),
+        settings.one_agreement,
+        ctrl_c_abort.clone(),
     ));
 
-    Arbiter::spawn(async move {
-        while let Some(agreement_id) = agreement_rx.next().await {
-            Arbiter::spawn(activity::spawn_activity(
-                api.clone(),
-                agreement_id,
-                exe_script.clone(),
-                commands_cnt,
-                activities.clone(),
-            ))
-        }
-    });
-
-    shutdown.await;
+    {
+        let activities = activities.clone();
+        let api = api.clone();
+        Arbiter::spawn(async move {
+            while let Some(agreement_id) = agreement_rx.next().await {
+                Arbiter::spawn(activity::spawn_activity(
+                    api.clone(),
+                    agreement_id,
+                    exe_script.clone(),
+                    commands_cnt,
+                    activities.clone(),
+                ));
+            }
+        });
+    }
+
+    match future_ctrl_c.await {
+        Ok(Ok(())) => log::info!("Caught ctrl-c"),
+        Ok(Err(error)) => log::error!("Failed listening to ctrl-c: {}", error),
+        Err(futures::future::Aborted) => log::info!("Listening to ctrl-c aborted"),
```

Comment:
would replace this technical info into some more meaningful:
```suggestion
        Err(futures::future::Aborted) => log::info!("`Finished processing one Agreement. Payment is confirmed. Finishing`"),
```